modules/home/Number.tsx

```diff
@@ -0,0 +1,41 @@
+'use client';
+import React, { useEffect, useRef } from 'react';
+import { animate, inView } from 'motion';
+
+const counter = (elementRef: HTMLDivElement, start: number, end: number) => {
+  let current = start;
+  let range = end - start;
+  let increment = end > start ? 1 : -1;
+  const step = Math.abs(Math.floor(1000 / range));
+  const timer = setInterval(() => {
+    current += increment;
+    elementRef.textContent = `${current}`;
+    if (current == end) {
+      clearInterval(timer);
+    }
+  }, step);
```

Comment:
the [animate](https://motion.dev/dom/animate) function could do this interpolation with more control over the duration and the timing function:
```suggestion
  const duration = Math.sqrt(Math.log(range)); // I tweaked that in the console so that the duration is not too different for all figures
  animate(
    (progress) => {
      current = Math.floor(start + progress * range);
      elementRef.textContent = `${current}`;
    },
    { duration , easing: 'ease-out' }
  )
```
I did not test it, it might need adjustments :)